compaction/table_state.hh

```diff
@@ -44,6 +45,7 @@ public:
     virtual future<> on_compaction_completion(sstables::compaction_completion_desc desc, sstables::offstrategy offstrategy) = 0;
     virtual bool is_auto_compaction_disabled_by_user() const noexcept = 0;
     virtual const tombstone_gc_state& get_tombstone_gc_state() const noexcept = 0;
+    virtual compaction_backlog_tracker& get_backlog_tracker() const noexcept = 0;
```

Comment:
I'm a little surprised by this. Shouldn't compaction_manager be responsible for creating backlog trackers?

This looks like something that could be dealt with internally.

Options:
1. I'm wrong
2. I'm right, but better to do it later
3. I'm right, and better to do it earlier

Reply:
Well, isn't it better for compaction_manager to keep track of the backlog trackers? That is, table_state doesn't know anything about it. compaction_manager calls compaction_strategy::make_backlog_tracker() and stores it, and doesn't need to bother compaction_group (through table_state) with it. It looks like something completely internal to compaction.

I'm not saying we should do it immediately, but let's agree on the direction.

Reply:
You're right. Backlog tracker is a concept that's confined to compaction layer and shouldn't be exposed to replica namespace. I'll rework the series to go in that direction, which is the correct one.